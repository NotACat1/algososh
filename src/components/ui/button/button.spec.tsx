import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Button } from './button';

describe('Button component', () => {
  test('renders button with text', () => {
    render(<Button text="Click me" />);
    const button = screen.getByRole('button', { name: /click me/i });
    expect(button).toMatchSnapshot();
  });

  test('renders button without text', () => {
    render(<Button />);
    const button = screen.getByRole('button');
    expect(button).toMatchSnapshot();
  });

  test('renders disabled button', () => {
    render(<Button disabled />);
    const button = screen.getByRole('button');
    expect(button).toBeDisabled();
  });

  test('renders loading indicator', () => {
    render(<Button isLoader />);
    const loaderIcon = screen.getByAltText('Загрузка.');
    expect(loaderIcon).toMatchSnapshot();
  });

  test('handles click event', () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick} />);
    const button = screen.getByRole('button');
    userEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
