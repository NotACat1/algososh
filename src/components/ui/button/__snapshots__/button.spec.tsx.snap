// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders button with text 1`] = `
<button
  class="text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    Click me
  </p>
</button>
`;

exports[`Button component renders button without text 1`] = `
<button
  class="text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button component renders loading indicator 1`] = `
<img
  alt="Загрузка."
  class="loader_icon"
  src="loader.svg"
/>
`;

exports[`Button component кнопки без текста 1`] = `
<button
  class="text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button component кнопки с текстом 1`] = `
<button
  class="text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    Click me
  </p>
</button>
`;
